### DISCUSSION TITLE
Why does it need screen recording access

### DISCUSSION THREAD

**Discussion by kambanwait** [jordanbaird/Ice#59] [General]
I couldn't see anything on here about the permissions, so thought I'd ask.

Why does it need screen recording permission and what does it do when it has the permission?

**Comment by jordanbaird** [comment]
To apply styling to the menu bar, Ice needs to capture the desktop background. The only thing it does with this is to overlay it on top of the menu bar when the appearance settings call for it (i.e. when using the "Split" shape kind). The screen itself is not actually recorded, but macOS requires the permission regardless.

**Reply by bobbd2000** [reply]
Does Ice still need the Screen Record permission if I won't be using any menu styling?

**Reply by jordanbaird** [reply]
At the moment, yes, but I'm working on making it so it only needs permissions if you choose to enable styling.

**Comment by bobbd2000** [comment]
Thanks
Bob

**Comment by jordanbaird** [comment]
Closing this, as there are now additional reasons for requiring screen recording permissions (outlined in #362). Follow there and #75 for more updates.
